doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,146 @@
+"collision" locations, and in each "collision" location a set of bitarrays that
+correspond to that subregion.
+
+When reading data, yt will identify which "coarse" index regions are necessary
+to read.  If any of those coarse index regions are covered by more than one
+file, it will examine the "fine" index for those regions and see if it is able
+to subset more efficiently.  Because all of these operations can be done with
+logical operations, this considerably reduces the amount of data that needs to
+be read from disk before expensive selection operations are conducted.
+
+For those situations that involve particles with regions of influence -- such
+as smoothed particle hydrodynamics, where particles have associated smoothing
+lenghts -- these are taken into account when conducting the indexing system.
+
+Efficiency of Index Orders
+--------------------------
+
+What this can lead to, however, is situations where (particularly at the edges
+of regions populated by SPH particles) the indexing system identifies
+collisions, but the relatively small number of particles and correspondingly
+large "smoothing lengths" result in a large number of "fine" index values that
+need to be set.
+
+Counterintuitively, this actually means that occasionally the "fine" indexing
+process can take an inordinately long amount of time for *small* datasets,
+rather than large datasets.
+
+In these situations, it is typically sufficient to set the "fine" index order
+to be much lower than its default value.  For instance, setting the
+``index_order`` to (5, 3) means that the full domain will be subdivided into 32
+bins in each dimension, and any "collision" zones will be further subdivided
+into 8 bins in each dimension (corresponding to an effective 256 bins across
+the full domain).
+
+If you are experiencing very long index times, this may be a productive
+parameter to modify.
+
+Index Caching
+-------------
+
+The index values are cached between instantiation, in a sidecar file named with
+the name of the dataset file and the suffix ``.indexII_JJ.ewah``, where ``II``
+and ``JJ`` are ``index_order1`` and ``index_order2``.  So for instance, if
+``index_order`` is set to (5, 7), and you are loading a dataset file named
+"snapshot_200.hdf5", after indexing, you will have an index sidecar file named
+``snapshot_200.hdf5.index5_7.ewah``.  On subsequent loads, this index file will
+be reused, rather than re-generated.
+
+The filename scheme can be changed by supplying an alternate filename to the
+``load`` command with the argument ``index_filename``.  For instance, if you
+are accessing data in a read-only location, you can specify that the index will
+be cached in a location that is write-accessible to you.
+
+These files contain the *compressed* bitmap index values, along with some
+metadata that describes the version of the indexing system they use and so
+forth.  If the version of the index that yt uses has changed, they will be
+regenerated.
```

Comment:
a question for us devs: is this determined by actually checking the `yt.__version__` constant, or is anything else taken into account. I worry that if only the version is compared, it might be a delicate process to implement changes there in between versions as it would break existing ewah files, likely without a clear error/warning.

Reply:
It's already checked internally; it's not necessarily related to the yt version, but instead to the version of the file format.  Right now an `OSError` will be raised internally in `particle_oct_container.pyx`, which gets cause in `particle_geometry_handler.py` and this message is displayed before it continues:

```        if ver != _bitmask_version:
            raise OSError("The file format of the index has changed since "
                          "this file was created. It will be replaced with an "
                          "updated version.")
```